build-scripts/MSVC.cmake

```diff
@@ -0,0 +1,56 @@
+if (NOT $ENV{DevEnvDir})
+ set(ENV{DevEnvDir} "C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE")
+endif()
+set(CMAKE_MAKE_PROGRAM $ENV{DevEnvDir}/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe CACHE PATH "")
+set(CMAKE_CXX_FLAGS_INIT "\
+/MP /utf-8 /bigobj /permissive- /sdl- /FC /Gd /GS- /Gy \
+/wd4068 /wd4146 /wd4819 /wd6237 /wd6319 /wd26444 /wd26451 /wd26495 /WX- /W1 \
+/TP /Zc:forScope /Zc:inline /Zc:wchar_t"
+)
+# /MP     # cl.exe build with multiple processes
+# /utf-8  # set source and execution character sets to UTF-8
+# /bigobj # increase # of sections in object files
+# /permissive- # to allow alternative operators ("and", "or", "not")
+# /sdl-   # disable additional security checks
+# /FC     # full path in compiler messages
+# /Gd     # __cdecl
+# /GS-    # disable buffer security checks
+# /Gy     # Enable Function-Level Linking
+# /wd4068 # unknown pragma
+# /wd4146 # negate unsigned
+# /wd4819 # codepage?
+# /wd6237 # short-circuit eval
+# /wd6319 # a, b: unused a
+# /wd26444 # unnamed objects
+# /wd26451 # overflow
+# /wd26495 # uninitialized member
+# /WX- # (do not) Treat Warnings as Errors
+# /W1  # Warning Level
+# /TP  # every file is a C++ file
+# /Zc:forScope  # Force Conformance in for Loop Scope
+# /Zc:inline    # Remove unreferenced COMDAT
+# /Zc:wchar_t   # wchar_t Is Native Type
+set(CMAKE_CXX_FLAGS_RELWITHDEBINFO_INIT
+"/Oi"
+)
+# /Ob1 # Inline Function Expansion (1 = only when marked as such)
+# /Oi  # Generate Intrinsic Functions
+add_compile_definitions(
+    _SCL_SECURE_NO_WARNINGS
+    _CRT_SECURE_NO_WARNINGS
+    WIN32_LEAN_AND_MEAN
+    LOCALIZE
+    USE_VCPKG
+)
+add_link_options(
+    /LTCG:OFF
+    /MANIFEST:NO
+)
```

Comment:
`/GF` and `/Gy` are useless without `/OPT:REF` and `/OPT:ICF`, are those defaulted? We should enable them unconditionally because they might not run for debug builds by default but there's no need to omit them in debug (and makes linking debug way slower).

Reply:
For the record, without `/MANIFEST:NO` the build system attempts to override the manifest and result in a linker error for duplicate resource.

Reply:
> `/GF` and `/Gy` are useless without `/OPT:REF` and `/OPT:ICF`, are those defaulted?

No. Added with f4a9bc4